virttest/utils_v2v.py

```diff
@@ -1146,30 +1181,77 @@ def _v2v_post_cmd():
             options = options + ' ' + rhv_upload_opts
         if opts_extra:
             options = options + ' ' + opts_extra
+        # Add -oo rhv-disk-uuid if '--no-copy' exists
+        if '-o rhv-upload' in options and '--no-copy' in options and '--oo rhv-disk-uuid' not in options:
+            for i in range(int(params.get('_disk_count', 0))):
+                options += ' -oo rhv-disk-uuid=%s' % str(uuid.uuid4())
 
-        # Construct a final virt-v2v command
-        cmd = '%s %s' % (V2V_EXEC, options)
+        # Construct a final virt-v2v command and remove redundant blanks
+        cmd = ' '.join(('%s %s' % (V2V_EXEC, options)).split())
         # Old v2v version doesn't support '-ip' option
         if not v2v_supported_option("-ip <filename>"):
             cmd = cmd.replace('-ip', '--password-file', 1)
-        cmd_result = process.run(cmd, timeout=v2v_cmd_timeout,
-                                 verbose=True, ignore_status=True)
+        # Save v2v command to params, then it can be passed to
+        # import_vm_to_ovirt
+        global_params.update({'v2v_command': cmd})
+
+        if not cmd_only:
+            cmd_result = process.run(cmd, timeout=v2v_cmd_timeout,
+                                     verbose=True, ignore_status=True)
     finally:
-        target_obj.cleanup()
+        if auto_clean:
+            target_obj.cleanup()
+        else:
+            # Save it into global params and release it by users
+            v2v_dirty_resources = global_params.get('v2v_dirty_resources', [])
+            v2v_dirty_resources.append(target_obj)
+            global_params.update({'v2v_dirty_resources': v2v_dirty_resources})
+        # Post-process for v2v
+        _v2v_post_cmd()
 
-    # Post-process for v2v
-    _v2v_post_cmd()
+    if cmd_only:
+        return cmd
 
     cmd_result.stdout = results_stdout_52lts(cmd_result)
     cmd_result.stderr = results_stderr_52lts(cmd_result)
 
     return cmd_result
 
 
+def cmd_run(cmd, obj_be_cleaned=None, auto_clean=True, timeout=18000):
+    """
+    Run v2v command and cleanup resources automatically
+
+    When you preserved the resources in v2v_cmd, the users need to release
+    them. This function can help you do it automatically.
+
+    :param obj_be_cleaned: the resources to be cleaned up
+    :param auto_clean: If true, cleanup obj_be_cleaned automatically
+    :param timeout: the timeout value for the command to run
+    """
+    try:
+        cmd_result = process.run(
+            cmd,
+            timeout=timeout,
+            verbose=True,
+            ignore_status=True)
```

Comment:
is it your expect that you want to ignore command result?

Reply:
Do you mean ignore_status?  all status will be saved in cmd_result, v2v will check that if need. This is same as the processing in v2v_cmd.